leakcanary-android-core/src/main/java/leakcanary/internal/RequestStoragePermissionActivity.kt

```diff
@@ -75,8 +75,8 @@ internal class RequestStoragePermissionActivity : Activity() {
     fun createPendingIntent(context: Context): PendingIntent {
       val intent = Intent(context, RequestStoragePermissionActivity::class.java)
       intent.flags = FLAG_ACTIVITY_NEW_TASK or FLAG_ACTIVITY_CLEAR_TOP
-      val flags = if (Build.VERSION.SDK_INT >= 23) {
-        FLAG_UPDATE_CURRENT or FLAG_IMMUTABLE
+      val flags = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+        FLAG_MUTABLE
```

Comment:
I'm confused, what's this change supposed to do?

Did you confirm that the crash still happens on the latest master? This crash is supposed to be already fixed by #2124, and we haven't made a new release since.

Reply:
Hi @pyricau,

I do test on the latest snapshot I can find and I am afraid yes still happen.
With FLAG_IMMTABLE you will be not able to update the current PendingIntent.
Basically, You can Still use FLAG_UPDATE_CURRENT  even if from 23 you have IMMUTABLE, but from 31 you must use IMMUTABLE or MUTABLE, in this case, if you want to still update the PendingIntent from 31, on my understanding, FLAG_MUTABLE is as FLAG_UPDATE_CURRENT.

Please correct me if I am wrong.